src/window/window_win.cpp

```diff
@@ -427,6 +417,47 @@ void create_window(sys::state& game_state, creation_parameters const& params) {
 	}
 }
 
+void change_cursor(sys::state const& state, cursor_type type) {
+	auto root = simple_fs::get_root(state.common_fs);
+	auto gfx_dir = simple_fs::open_directory(root, NATIVE("gfx"));
+	auto cursors_dir = simple_fs::open_directory(gfx_dir, NATIVE("cursors"));
+
+	if(state.win_ptr->cursors[uint8_t(type)] == HCURSOR(NULL)) {
+		native_string_view fname = NATIVE("normal.cur");
+		switch(type) {
+		case cursor_type::normal:
+			fname = NATIVE("normal.cur");
+			break;
+		case cursor_type::busy:
+			fname = NATIVE("busy.ani");
+			break;
+		case cursor_type::drag_select:
+			fname = NATIVE("dragselect.ani");
+			break;
+		case cursor_type::hostile_move:
+			fname = NATIVE("attack_move.ani");
+			break;
+		case cursor_type::friendly_move:
+			fname = NATIVE("friendly_move.ani");
+			break;
+		case cursor_type::no_move:
+			fname = NATIVE("no_move.ani");
+			break;
+		default:
+			fname = NATIVE("normal.cur");
+			break;
+		}
+		if(auto f = simple_fs::peek_file(cursors_dir, fname); f) {
+			auto path = simple_fs::get_full_name(*f);
+			state.win_ptr->cursors[uint8_t(type)] = LoadCursorFromFileW(path.c_str()); //.cur or .ani
+		} else {
+			return; //can't load
```

Comment:
there is an issue here: if the cursor file is missing, then your code will hit the file system each frame looking for it, which will be extremely slow. You need to instead store a default cursor or some other indicator not to go looking for it.

Reply:
Ok schowomby

Reply:
fixed